### ISSUE TITLE
Search elliptic curves over number fields by rank

### ISSUE THREAD

**Issue by jwj61** [LMFDB/lmfdb#799]
There is no search box for rank for elliptic curves over number fields.

**Comment by jenpaulhus** [comment]
JEC says we are waiting for all ranks to be computed.

**Comment by JohnCremona** [comment]
I will get back to computing ranks once I have finished, or almost finished, finding curves.  So far this is almost done except for totally real sextic fields (where "done" means we have curves for all Hilbert or Bianchi newforms which we have).  It is likely to be a major job computing ranks for about 350,000 curves over fields of degree up to 6, so it will be hard to decide at what point we can offer such a search without it being misleading on account of many curves not kowing their rank.

**Comment by AndrewVSutherland** [comment]
@JohnCremona if the rank computation is just a matter of throwing cpu cycles at it, let me know -- I have a few machines I could contribute to the cause (and/or we could use GCE)

**Comment by JohnCremona** [comment]
The script I was using for the IWFs (which is in scripts/ecnf/point-find.py) is just an interface to Magma's MordellWeilShaInformation() + code to transfer points between curves in each isogeny class and saturate.  So it needs to be run a machine with Magma and with database read/write access.  Of course it would be possible to separate this into 3 stages (database read to prepare a Magma input file, run Magma, parse output) but still the main cpu-intensive part needs Magma access.

**Comment by AndrewVSutherland** [comment]
I have a 48 core machine with magma on it that I could use.

**Comment by AndrewVSutherland** [comment]
@JohnCremona is it still your plan to complete #610 before starting this?  Would it make sense to go ahead with the computations for the fields of degree up to 5?

FYI, over the next few weeks we will be moving the postgres database and beta.lmfdb.org to a brand new machine that arrived just last week.  This will leave legendre.mit.edu completely free for LMFDB development, and we could easily run these computations there (legendre has 48 cores, the latest version of magma, and will have a 1 gigabit connection to the machine hosting the postgres database).

**Comment by JohnCremona** [comment]
@AndrewVSutherland No, there is no need to wait for #610.  This just dropped off my list of current activities.  Despite what I wrote here 2 years ago, there is no need for database access and the whole thing can be done offline, reading curves from plain text data files and outputting the rank information and points ditto (using the text files as in the repository ecnf-data, possibly adjusted).  I have plenty (>300) of cores on machines with Magma which can run that, with no need for database access until the new data is uploaded.  Thanks for the offer anyway.

Of course it remains to be seen how many ranks we can actually compute (and not just bounds) in reasonable time for the higher degree fields.  No-one has every done this for hundreds of curves over fields of degree>2!

**Comment by JohnCremona** [comment]
This will soon be possible, as I have computed all BSD data for curves over fields of degree 2,3,4, and partial data (without analytic ranks which are still running) in degrees 5 and 6.  I'll make a separate issue for that.

**Comment by JohnCremona** [comment]
Closing as most curves now have ranks, and the search box has been inplemented
